crates/driver/src/tests/cases/jit_orders.rs

```diff
@@ -0,0 +1,209 @@
+use crate::{
+    domain::{
+        competition::{order, order::Side},
+        eth,
+    },
+    tests::{
+        self,
+        cases::{is_approximately_equal, EtherExt},
+        setup::{
+            self,
+            ab_adjusted_pool,
+            ab_liquidity_quote,
+            ab_order,
+            ab_solution,
+            test_solver,
+            SolverName,
+            Test,
+        },
+    },
+};
+
+struct Amounts {
+    sell: eth::U256,
+    buy: eth::U256,
+}
+
+struct Execution {
+    // The executed net-amounts (including network fee) reported by the solver
+    solver: Amounts,
+    // The executed net-amounts (including network and protocol) reported by the driver
+    driver: Amounts,
+}
+
+struct Order {
+    sell_amount: eth::U256,
+    buy_amount: eth::U256,
+    side: order::Side,
+}
+
+struct JitOrder {
+    order: Order,
+    solver: SolverName,
+}
+
+struct Solution {
+    solver_name: SolverName,
+    is_surplus_capturing_jit_order: bool,
+    expected_score: eth::U256,
+}
+
+struct TestCase {
+    order: Order,
+    execution: Execution,
+    jit_order: JitOrder,
+    solutions: Vec<Solution>,
```

Comment:
What's the reason to not have the `JitOrder` as part of the `Solution`?
Also if you can only configure a single jit order in the test case why do would we need `solutions` to be a vector? How would we ever return multiple different solutions? If this is just about differently named solvers returning the same solution which results in a different score because some are not surplus capturing jit order owners I think this should be handled differently.

Unless I'm overlooking some glaring complication I really think putting the jit orders directly into the solution a solver is supposed to return would resolve a lot of complexity of this PR.

Reply:
Additionally if the JIT order lives in the Solution itself I don't think there is a need for the `SolverName` enum. For testing the correctness we don't really have to have multiple solvers return solutions in the same test case. Instead of having a single test case where 3 solvers return some solution we could simply have 3 separate test cases where an unnamed solver returns a single solution.

Reply:
> why do would we need solutions to be a vector?

@MartinquaXD I want to open the possibility to extend the test suite with more JIT order related tests, since there aren't any. So, I wanted to make it possible to write several solutions with different JIT orders in it. But I can make it as a simple solution and if needed we can extend it later or if we do it by having different test cases as you mentioned.